### PR TITLE
formulae*: depend on libfuse on linux, deprecate only on mac

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Of course if one of these does not build anymore we can get rid of it. We might move some of these to be linux-only if needed. But for the time being I added the linux fuse dependencies so that we can build these on linux.

**Comment by carlocab** [comment]
Having formulae that are deprecated only on one OS is going to get weird after a while... 

For example, if we want to eventually disable some of these formulae for Mac, we'll need to fix the workflow that automatically deletes disabled formulae. I suppose that workflow would have to not delete partially disabled formulae, but instead make them `depends_on :linux` or `depends_on :macos` as needed.

That said, the workflow only opens PRs, so I suppose we could always modify the PR where they are deleted accordingly.

**Comment by iMichka** [comment]
Agreed. This is why I would like to gather a little bit more feedback from other homebrew-core maintainers.

The fuse formulae are the only formulae having this issue. I hope this situation is quite exceptional, I do not believe that we will have to deprecate formulae for a single platform on a regular basis.

**Comment by SMillerDev** [comment]
I guess we can move it to be linux-only instead of disabling.

File: Formula/archivemount.rb [hunk 1]
```diff
@@ -11,11 +11,17 @@ class Archivemount < Formula
     sha256 "428113b60673b6bb8be9467587f1d82bf4c9447c7f0bbdea47749bed3ec86798" => :high_sierra
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "pkg-config" => :build
   depends_on "libarchive"
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/archivemount.rb:16]:
```suggestion

```

File: Formula/btfs.rb [hunk 2]
```diff
@@ -13,13 +13,19 @@ class Btfs < Formula
...
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "pkg-config" => :build
   depends_on "libtorrent-rasterbar"
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/btfs.rb:20]:
```suggestion

```

File: Formula/dislocker.rb [hunk 3]
```diff
@@ -12,11 +12,17 @@ class Dislocker < Formula
     sha256 "2b1e50229eb344c432db6cc35fd42b6e91d713f97f81d6f5067087f5c59b6cb3" => :high_sierra
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "cmake" => :build
   depends_on "mbedtls"
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/dislocker.rb:17]:
```suggestion

```

File: Formula/encfs.rb [hunk 4]
```diff
@@ -16,13 +16,19 @@ class Encfs < Formula
...
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
   depends_on "gettext"
   depends_on "openssl@1.1"
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/encfs.rb:23]:
```suggestion

```

File: Formula/ext2fuse.rb [hunk 5]
```diff
@@ -12,10 +12,16 @@ class Ext2fuse < Formula
     sha256 "0b8e89292e91a8fbe00430ae16a3ebbfdbba1017f6dee4801bcf8e63d238962f" => :high_sierra
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "e2fsprogs"
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/ext2fuse.rb:16]:
```suggestion

```

File: Formula/ext4fuse.rb [hunk 6]
```diff
@@ -16,10 +16,16 @@ class Ext4fuse < Formula
     sha256 "b11f564b7e7c08af0b0a3e9854973d39809bf2d8a56014f4882772b2f7307ac1" => :yosemite
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "pkg-config" => :build
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/ext4fuse.rb:20]:
```suggestion

```

File: Formula/fuse-zip.rb [hunk 7]
```diff
@@ -16,11 +16,17 @@ class FuseZip < Formula
     sha256 "e72d442a43e1396c8a744e73bc9d197cbef7bb996bba97bff4b377c253c12ed8" => :high_sierra
   end
 
-  deprecate! date: "2020-11-10", because: "requires FUSE"
-
   depends_on "pkg-config" => :build
   depends_on "libzip"
-  depends_on :osxfuse
+
```
Comment (carlocab) [inline Formula/fuse-zip.rb:21]:
```suggestion

```

**Comment by jonchang** [comment]
> I guess we can move it to be linux-only instead of disabling.

I would be in favor of either doing this or just removing these formulae entirely and having the community maintain a homebrew-fuse tap. I believe FUSE on Linux requires a setuid root executable and I don't think we have a good way to test it FUSE implementations since you have to loosen some Docker security restrictions in order to get FUSE to work.

**Comment by gromgit** [comment]
> removing these formulae entirely and having the community maintain a homebrew-fuse tap

Funny you should mention that. I recently started the `gromgit/new-life` tap to host `unrar`, the new `macfuse` and `encfs`. Looking at the list of FUSE-enabled formulae, I'm seriously thinking of tapping `gromgit/fuse` just for them.

> I believe FUSE on Linux requires a setuid root executable

Yes, `libfuse` on Linux uses the setuid `fusermount` binary.

**Comment by iMichka** [comment]
> I guess we can move it to be linux-only instead of disabling.

Not all of these build on linux, so I can not just blindly migrate everything. They need to be tested/fixed/patched one by one and I don't have time for that right now.

Right now I would like to get this merged so that it brings the linuxbrew-core/homebrew-core code closer together, and fixes a few linux builds as a side-effect. The change makes the current state of things clearer for the users.

If you want to migrate some of these to a separate tap, delete a few others, or make some of these linux-only: can we discuss this in a another PR? Just to move forward, as the discussion about the future of these formulae might be discussed on a case-by-case basis.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**